main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  # my-each
+
+  def my_each
+    ind = 0
+
+    while ind != length
+      yield self[ind]
+      ind += 1
+    end
+
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    index = 0
+    my_each do |el|
+      yield(el, index)
+      index += 1
+    end
+
+    self
+  end
+
+  # my_select
+
+  def my_select
+    filtered_arr = []
+
+    while length != 0
+      yield(first) && filtered_arr.push(first)
+      shift
+    end
+
+    filtered_arr
+  end
+
+  # my_all
+
+  def my_all?
+    output = true
+
+    while length != 0
+      if !yield(first)
+        output = false
+        break
+      else
+        shift
+      end
+    end
+    output
+  end
+
+  # my_any
+
+  def my_any?
+    output = false
+    my_each do |el|
+      if yield el
+        output = true
+        break
+      end
+    end
+    output
+  end
+
+  # my_none
+
+  def my_none?
+    output = true
+
+    while length != 0
+      if yield first
+        output = false
+        break
+      else
+        shift
+      end
+    end
+    output
+  end
+
+  # my_count
+
+  def my_count
+    count = 0
+
+    my_each do |el|
+      yield(el) && count += 1
+    end
+    count
+  end
+
+  # my_map
+
+  def my_map(&proc_block)
+    new_arr = []
+
+    my_each { |el| new_arr.push(proc_block.call(el)) }
+
+    new_arr
+  end
+
+  # my_inject
+
+  def my_inject(num)
+    output = num
+
+    my_each do |sum|
+      output = yield(output, sum)
+    end
+    output
+  end
+end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- It should raise an LocalJumpError when no block is given. 
- It should take a block as an argument and apply it to each element of the object. Your's it's not recognizing it. 
- Your method should consider the possibility when an initial value is given.
- Your method should work with symbols as arguments (:+) and perform the right operation.
- Your method should work with ranges.